posts/2024-05-13-jakarta-nosql-challenge-switching-nosql-easily.adoc

```diff
@@ -0,0 +1,258 @@
+  <artifactId>jnosql-couchbase</artifactId>
+  <version>1.1.0</version>
+</dependency>
+----
+
+=== Configuring a Local Couchbase Cluster
+
+First of all, let's add the following content to the `docker-compose.yml` file:
+
+[source, yaml]
+----
+  couchbase-db:
+    image: couchbase
+    container_name: couchbase-db
+    ports:
+      - "8091-8097:8091-8097"
+      - "9123:9123"
+      - "11207:11207"
+      - "11210:11210"
+      - "11280:11280"
+      - "18091-18097:18091-18097"
+----
+
+After to apply these changes, initialize the container by executing the following command:
+
+[source, bash]
+----
+docker-compose up -d
+----
+
+Now, let's configure the Couchbase cluster for this study scenario:
+
+. With the Couchbase container started, go to: http://localhost:8091/ui/index.html
+. Select the "Configure New Cluster" option
++
+[.img_border_light]
+image::/img/blog/couchbase-01.png[Couchbase 01, align="center" width=80%]
+. Defines a name for the cluster name
+. Set `root` to "Administrator" username
+. Set `123456` as "Password"
++
+[.img_border_light]
+image::/img/blog/couchbase-02.png[Couchbase 02, align="center" width=80%]
+. Check “I accept the terms and conditions” and click the “Finish With Defaults” button
++
+[.img_border_light]
+image::/img/blog/couchbase-03.png[Couchbase 03, align="center" width=80%]
+. Go to the "Buckets" section
+. Create a `jnopo` Bucket by clicking on the "Add Bucket" option
++
+[.img_border_light]
+image::/img/blog/couchbase-04.png[Couchbase 04, align="center" width=80%]
+. Click on "Scopes & Collections" and then click on "Add Collection"
+. Enter `GameMatch` as the collection name and click the "Save" button to complete your creation
++
+[.img_border_light]
+image::/img/blog/couchbase-05.png[Couchbase 05, align="center" width=80%]
+. Now, in the "Query" section, run this command https://www.couchbase.com/products/n1ql/[N1QL] to create the primary index for the `GameMatch` collection:
++
+[source, n1ql]
+----
+CREATE PRIMARY INDEX primaryGameMatch ON jnopo._default.GameMatch
+----
++
+[.img_border_light]
+image::/img/blog/couchbase-06.png[Couchbase 06, align="center" width=80%]
+
+With a Couchbase cluster running, we just need to set up Eclipse JNoSQL to connect to it.
+
+Apply the following changes into the `resources/META-INF/microprofile-config.properties` file:
+
+[source, properties]
+----
+## Couchbase
+jnosql.couchbase.host=couchbase://localhost
+jnosql.couchbase.user=root
+jnosql.couchbase.password=123456
+jnosql.document.provider=org.eclipse.jnosql.databases.couchbase.communication.CouchbaseDocumentConfiguration
+----
+
+The `jnosql.document.provider` property is required in this context because we have more than one Eclipse JNoSQL Database API for Document NoSQL implementations. Learn more about these configurations about Eclipse JNoSQL Database API for Couchbase at this https://github.com/eclipse/jnosql-databases/tree/main?tab=readme-ov-file#configuration-2[Git repository].
```

Comment:
```suggestion
The `jnosql.document.provider` property is required in this context because we have more than one Eclipse JNoSQL Database API for Document NoSQL implementations. You can learn more about the configurations available for Eclipse JNoSQL Database API for Couchbase in this https://github.com/eclipse/jnosql-databases/tree/main?tab=readme-ov-file#configuration-2[Git repository].
```